python/lsst/ts/m2gui/controltab/tab_actuator_control.py

```diff
@@ -33,7 +57,516 @@ class TabActuatorControl(TabDefault):
         Table's title.
     model : `Model`
         Model class.
+
+    Attributes
+    ----------
+    model : `Model`
+        Model class.
     """
 
+    MAX_DISPLACEMENT = 10**7
+
     def __init__(self, title, model):
         super().__init__(title, model)
+
+        self._info_script = {
+            "file": create_label(""),
+            "progress": self._create_progress_bar(0, 100, 0),
+        }
+
+        self._labels_force = {
+            "axial_min": create_label(""),
+            "axial_max": create_label(""),
+            "axial_total": create_label(""),
+            "tangent_min": create_label(""),
+            "tangent_max": create_label(""),
+            "tangent_total": create_label(""),
+        }
+
+        self._buttons_script = {
+            "load_script": set_button("Load Script", self._callback_script_load_script),
+            "run": set_button("Run", self._callback_script_command, CommandScript.Run),
+            "stop": set_button(
+                "Stop", self._callback_script_command, CommandScript.Stop
+            ),
+            "pause": set_button(
+                "Pause", self._callback_script_command, CommandScript.Pause
+            ),
+            "resume": set_button(
+                "Resume", self._callback_script_command, CommandScript.Resume
+            ),
+            "clear": set_button(
+                "Clear", self._callback_script_command, CommandScript.Clear
+            ),
+        }
+
+        self._target_displacement = self._create_target_displacement()
+        self._displacement_unit_selection = self._create_displacement_unit_selection()
+
+        self._ring_selection = self.create_combo_box_ring_selection()
+        self._buttons_actuator_selection = self._create_buttons_actuator_selection()
+        self._buttons_actuator_selection_support = {
+            "select_ring": set_button("Select Ring", self._callback_select_ring),
```

Comment:
Command to Clear Ring might be handy. Also, I prefer to say "Clear All" or "Unselect All" instead of "Clear Selection" - "Clear Selection" doesn't provide if that's a ring or all selected.

Reply:
And see above for ButtonGrid, you really don't need hash/map for that.